src/Packetery/Module/Order/LabelPrint.php

```diff
@@ -283,24 +284,40 @@ public function outputLabelsPdf(): void {
 
 			$redirectTo = $this->httpRequest->getQuery( PacketActionsCommonLogic::PARAM_REDIRECT_TO );
 			if ( PacketActionsCommonLogic::REDIRECT_TO_ORDER_DETAIL === $redirectTo && null !== $idParam ) {
-				$this->packetActionsCommonLogic->redirectTo( $redirectTo, $this->orderRepository->getById( (int) $idParam, true ) );
+				$this->packetActionsCommonLogic->redirectTo( $redirectTo, $this->orderRepository->getById( $idParam, true ) );
 			}
 			$this->packetActionsCommonLogic->redirectTo( PacketActionsCommonLogic::REDIRECT_TO_ORDER_GRID );
 			return;
 		}
 
 		foreach ( $packetIds as $orderId => $packetId ) {
-			$wcOrder = $this->orderRepository->getWcOrderById( (int) $orderId );
-			if ( null === $wcOrder ) {
+			try {
+				$order = $this->orderRepository->getById( $orderId );
+			} catch ( Module\Exception\InvalidCarrierException $exception ) {
 				continue;
 			}
 
-			if ( $response instanceof Response\PacketsLabelsPdf ) {
-				$wcOrder->add_order_note( __( 'Packeta: Label has been created.', 'packeta' ) );
+			$wcOrder = $this->orderRepository->getWcOrderById( $orderId );
+			if ( null === $wcOrder || null === $order ) {
+				continue;
 			}
 
+			$link = Html::el( 'a' )
```

Comment:
vyčlenil bych do samostatné metody v jiné třídě.  nebudeš tak opakovat stejný kód na několika místech.  také bych to nepojmenoval $link, protože v proměnné nemáš url.

Reply:
Vycleneno a zapracovano.